xen/arch/arm/tee/ffa_partinfo.c

```diff
@@ -65,11 +65,22 @@ int32_t ffa_handle_partition_info_get(uint32_t w1, uint32_t w2, uint32_t w3,
                                       uint32_t *count, uint32_t *size)
 {
     int32_t ret;
-    struct domain *d = current->domain;
+    struct domain *dom, *d = current->domain;
     struct ffa_ctx *ctx = d->arch.tee;
     uint32_t src_size, dst_size;
     void *dst_buf;
-    uint32_t num_ffa_sp;
+    uint32_t num_ffa_vm = 0, num_ffa_sp;
+
+    /* Count the number of VM with FF-A support */
```

Comment:
Consider moving this to the if-block where we're counting the SPs below.

Reply:
I need this value later to asses if there is enough space in the receive buffer